Vostok.Metrics.System/Host/DiskUsageCollector_Linux.cs

```diff
@@ -0,0 +1,115 @@
+    {
+        private Dictionary<string, DiskStats> previousDisksStatsInfo = new Dictionary<string, DiskStats>();
+        private readonly Stopwatch stopwatch = new Stopwatch();
+
+        public void Collect(HostMetrics metrics, IEnumerable<string> diskstatsLines)
+        {
+            var deltaTime = (double) stopwatch.ElapsedTicks / Stopwatch.Frequency;
+
+            var disksUsageInfo = new Dictionary<string, DiskUsageInfo>();
+
+            foreach (var diskStats in ParseDiskstats(diskstatsLines))
+            {
+                var result = new DiskUsageInfo {DiskName = diskStats.DiskName};
+
+                if (previousDisksStatsInfo.ContainsKey(diskStats.DiskName))
+                    FillInfo(result, previousDisksStatsInfo[diskStats.DiskName], diskStats, deltaTime);
+
+                disksUsageInfo[result.DiskName] = result;
+
+                previousDisksStatsInfo[diskStats.DiskName] = diskStats;
+            }
+
+            metrics.DisksUsageInfo = disksUsageInfo;
+
+            RemoveUnusedEntries(disksUsageInfo);
+
+            stopwatch.Restart();
+        }
+
+        private void FillInfo(DiskUsageInfo toFill, DiskStats previousDiskStats, DiskStats diskStats, double deltaTime)
+        {
+            var readsDelta = diskStats.ReadsCount - previousDiskStats.ReadsCount;
+            var writesDelta = diskStats.WritesCount - previousDiskStats.WritesCount;
+            var timeReadDelta = diskStats.TimeSpentReading - diskStats.TimeSpentReading;
+            var timeWriteDelta = diskStats.TimeSpentWriting - diskStats.TimeSpentWriting;
+
+            if (readsDelta > 0)
+                toFill.ReadLatency = (double) timeReadDelta / readsDelta;
+            if (writesDelta > 0)
+                toFill.WriteLatency = (double) timeWriteDelta / writesDelta;
+
+            toFill.CurrentQueueLength = diskStats.CurrentQueueLength - previousDiskStats.CurrentQueueLength;
+
+            toFill.ReadsPerSecond = readsDelta / deltaTime;
+            toFill.WritesPerSecond = writesDelta / deltaTime;
+
+            toFill.IdleTimePercent = ((1 - (timeReadDelta + timeWriteDelta) / deltaTime) * 100).Clamp(0, 100);
+        }
+
+        private void RemoveUnusedEntries(Dictionary<string, DiskUsageInfo> disksUsageInfo)
+        {
+            previousDisksStatsInfo = previousDisksStatsInfo
+               .Where(x => disksUsageInfo.ContainsKey(x.Key))
+               .ToDictionary(x => x.Key, y => y.Value);
+        }
+
+        private List<DiskStats> ParseDiskstats(IEnumerable<string> diskstats)
+        {
+            var disksStats = new List<DiskStats>();
+
+            try
+            {
+                foreach (var line in diskstats)
+                {
+                    if (FileParser.TrySplitLine(line, 14, out var parts) && parts[1] == "0")
+                    {
+                        var stats = new DiskStats {DiskName = parts[2]};
+
+                        if (long.TryParse(parts[4], out var readsCount))
```

Comment:
According to the [doc](https://www.kernel.org/doc/Documentation/ABI/testing/procfs-diskstats), the value at this index (zero-based 4) returns "reads merged". The value we want is "reads completed" at index 3, right?